src/stability.md

```diff
@@ -49,4 +59,67 @@ are nondeterministic and often unknown at compile-time.
 Always ping @oli-obk, @RalfJung, and @Centril if you are adding more `allow_internal_unstable`
 attributes to any `const fn`
 
+## staged_api
+
+Any crate that uses the `stable`, `unstable`, or `rustc_deprecated` attributes
+must include the `#![feature(staged_api)]` attribute on the crate.
+
+## rustc_deprecated
+
+The deprecation system shares the same infrastructure as the stable/unstable
+attributes. The `rustc_deprecated` attribute is similar to the [`deprecated`
+attribute]. It was previously called `deprecated`, but was split off when
+`deprecated` was stabilized. The `deprecated` attribute cannot be used in a
+`staged_api` crate, `rustc_deprecated` must be used instead. The deprecated
+item must also have a `stable` or `unstable` attribute.
+
+`rustc_deprecated` has the following form:
+
+```rust,ignore
+#[rustc_deprecated(
+    since = "1.38.0",
+    reason = "explanation for deprecation",
+    suggestion = "other_function"
+)]
+```
+
+The `suggestion` field is optional. If given, it should be a string that can
+be used as a machine-applicable suggestion to correct the warning. This is
+typically used when the identifier is renamed, but no other significant
+changes are necessary.
+
+Another difference from the `deprecated` attribute is that the `since` field
+is actually checked against the current version of `rustc`. If `since` is in a
+future version, then the `deprecated_in_future` lint is triggered which is
+default `allow`, but most of the standard library raises it to a warning with
+`#![warn(deprecated_in_future)]`.
+
+[`deprecated` attribute]: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute
+
+## -Zforce-unstable-if-unmarked
+
+The `-Zforce-unstable-if-unmarked` flag has a variety of purposes to help
+enforce that the correct crates are marked as unstable, but can still use
```

Comment:
I don't understand this sentence. What do you mean by "can still use private crates"

Reply:
Maybe an example would help:

`chalk-engine` gets marked as unstable `rustc_private`.  `chalk-engine` depends on `chalk-macros` which also gets marked as unstable `rustc_private`.  Normally it would not be able to access the `chalk-macros` crate without the `#![feature(rustc_private)]` flag, but that is not something most crates.io crates have. So [this code](https://github.com/rust-lang/rust/blob/58b54911fabcd1b328fab78014b6833a4e8cfaa3/src/librustc/middle/stability.rs#L746-L759) ensures that once something has been "forced" unstable, it can also access other forced unstable crates.

I pushed a change that tries to clarify it, but it still seems a little awkward.